File: ccc/2014/s3/ccc2014s3.cpp
#include <stdio.h>
#include <stack>
#include <iostream>
using namespace std;

int T = 0;
int N;

int main() {
    cin >> T;
    for (int _=0; _<T; _++) {
        cin >> N;
        stack<int> mountain;
        stack<int> branch;
        for (int __=0; __<N; __++) {
            int n;
            cin >> n;
            mountain.push(n);
        }
        int n = 1;
        int status = 0;
        while (1) {
            if (mountain.size() > 0 && mountain.top() == n) {
                n++;
                mountain.pop();
            }
            else if (branch.size() > 0 && branch.top() == n) {
                n++;
                branch.pop();
            }
            else if (mountain.size() > 0 && branch.size() == 0) {
                branch.push(mountain.top());
                mountain.pop();
            }
            else if (mountain.size() == 0 && branch.size() > 0) {
                status = 1;
                break;
            }
            else if (mountain.size() > 0 && branch.size() > 0) {
                if (mountain.top() < branch.top()) {
                    branch.push(mountain.top());
                    mountain.pop();
                }
                else {
                    status = 1;
                    break;
                }
            }
            else {
                break;
            }
        }
        cout << (status ? "N" : "Y") << "\n";
    }

    return 0;
}

File: ccc/2014/s3/ccc2014s3.py
T = int(input())

for _ in range(T):
    N = int(input())
    mountain = [int(input()) for _ in range(N)]
    branch = []
    out = "Y"
    n = 1
    while True:
        if len(mountain) > 0 and len(branch) > 0:
            if mountain[-1] == n:
                n += 1
                mountain.pop()
            elif branch[-1] == n:
                n += 1
                branch.pop()
            else:
                if mountain[-1] < branch[-1]:
                    branch.append(mountain.pop())
                else:
                    out = "N"
                    break
        elif len(mountain) > 0:
            if mountain[-1] == n:
                n += 1
                mountain.pop()
            else:
                branch.append(mountain.pop())
        elif len(branch) > 0:
            if branch[-1] == n:
                n += 1
                branch.pop()
            else:
                out = "N"
                break
        else:
            break
    print(out)